docs/docs/reference/built-in-components/gatsby-plugin-image.md

```diff
@@ -0,0 +1,129 @@
+---
+title: Gatsby Image plugin
+---
+
+This guide will show you how to configure your images, including choosing layouts, placeholders and image processing options. If you would like to learn how to set up the image plugins and use images on your site, see [the how-to guide](/docs/how-to/images-and-media/using-gatsby-plugin-image).
+
+## Components
+
+The Gatsby Image plugin includes two components to display responsive images on your site, used for static and dynamic images.
+
+- **`StaticImage`:** Use this if the image is the same every time the component is used. _Examples: site logo, index page hero image_
+- **`GatsbyImage`:** Use this if the image is passed-in to the component as a prop. _Examples: Blog post hero image, author avatar_
+
+## Image options
+
+The options for both components are almost the same, with the difference being how you specify them. When using `StaticImage`, they are passed as props to the component, whereas for the `GatsbyImage` component they are passed to the `gatsbyImageData` GraphQL resolver. There is a small change in the values too: in the `StaticImage` component, props such as layout and placeholder take a string, while the resolver takes a GraphQL enum, which is in upper case by convention and is not quoted like a string. Both syntaxes are shown in the reference below.
+
+> It is a very good idea to use [the GraphiQL IDE](/docs/how-to/querying-data/running-queries-with-graphiql) when writing your `gatsbyImageData` queries. It includes auto-complete and inline documentation for all of the options and lets you see the generated image data right inside the IDE.
+
+### Layout
+
+**Prop name: `layout`**
+
+The image components support three types of layout, which determine the image sizes that are generated, as well as the resizing behavior of the image itself in the browser. You can compare the layouts in the video below:
+
+<video controls autoplay loop>
+  <source type="video/mp4" src="./layouts.mp4" />
+  <p>Your browser does not support the video element.</p>
+</video>
+
+To set the layout, pass in the type to the layout prop. e.g.
+
+```jsx
+<StaticImage
+  src="./dino.png"
+  alt="A dinosaur"
+  // highlight-next-line
+  layout="fixed"
+/>
+```
+
+Alternatively, pass it to the resolver:
+
+```graphql
+dino {
+  childImageSharp {
+    # highlight-next-line
+    gatsbyImageData(layout: FIXED)
+  }
+}
+```
+
+#### Constrained
```

Comment:
Throwing this out there as an idea: What do you think about putting these values into a table instead of paragraphs? Trying to make this page easier to scan.

Something like: 

| Name | Component prop | Resolver prop | Description |
| --- | --- | --- | --- |
| Constrained | `"constrained"` | `CONSTRAINED` | This is the default layout. It displays... |

I'm imagining something similar to this vibe: https://material-ui.com/api/radio/

Reply:
Or actually, maybe that makes more sense for the table of contents at the beginning of the doc (~ line 19), since the descriptions aren't so long.

Then it could be:

| Prop | Value | Default Value | Description |
| --- | --- | --- | --- |
| `layout` | constrained or fixed or full width | constrained | ... |

Thoughts?